pinot-core/src/main/java/org/apache/pinot/core/query/selection/SelectionOperatorUtils.java

```diff
@@ -339,6 +339,74 @@ public static DataTable getDataTableFromRows(Collection<Object[]> rows, DataSche
     return dataTableBuilder.build();
   }
 
+  /**
+   * The default value for each column type.
+   */
+  public static Serializable getDefaultValue(ColumnDataType dataType) {
+    switch (dataType) {
+      // Single-value column
+      case INT:
+        return THREAD_LOCAL_INT_FORMAT.get().format(((Number) 0).intValue());
+      case LONG:
+        return THREAD_LOCAL_LONG_FORMAT.get().format(((Number) 0).longValue());
+      case FLOAT:
+        return THREAD_LOCAL_FLOAT_FORMAT.get().format(((Number) 0).floatValue());
+      case DOUBLE:
+        return THREAD_LOCAL_DOUBLE_FORMAT.get().format(((Number) 0).doubleValue());
```

Comment:
```java
      case INT:
         return THREAD_LOCAL_INT_FORMAT.get().format(0);
       case LONG:
         return THREAD_LOCAL_LONG_FORMAT.get().format(0L);
       case FLOAT:
         return THREAD_LOCAL_FLOAT_FORMAT.get().format(0F);
       case DOUBLE:
         return THREAD_LOCAL_DOUBLE_FORMAT.get().format(0D);
```

Reply:
Any comment?

Reply:
There’s no need to box 0 to get the right primitive type, I guess you’re doing this to make the type unambiguous, but you can do that with literal suffixes

Reply:
Done